### PR TITLE
mastercontainer - use Caddy for generating self-singed cert

### PR DESCRIPTION
**Pull request by szaimen**
Alternative approach to https://github.com/nextcloud/all-in-one/pull/3069
Supercedes https://github.com/nextcloud/all-in-one/pull/3069

### CODE REVIEW

File: Containers/mastercontainer/Caddyfile [hunk 1]
```diff
@@ -21,7 +21,14 @@
...
 
 http://:80 {
-  redir https://{host}{uri}
+    redir https://{host}{uri}
+}
+
+https://:8080 {
+    reverse_proxy localhost:8000
+    tls internal {
+        on_demand
```
Comment (szaimen) [inline Containers/mastercontainer/Caddyfile:30]: This needs double check if also ip-addresses get accepted here...

**Comment by szaimen** [comment]
@Zoey2936 would this approach be fine with you?

File: Containers/mastercontainer/Caddyfile (hunk 1 above)
Comment (Zoey2936) [reply Containers/mastercontainer/Caddyfile:30]: yes

**Comment by Zoey2936** [comment]
> @Zoey2936 would this approach be fine for you?

what is different to #3096?

File: Containers/mastercontainer/Caddyfile (hunk 1 above)
Comment (szaimen) [reply Containers/mastercontainer/Caddyfile:32]: should be done with https://github.com/nextcloud/all-in-one/pull/3416/commits/9a0fd6d0a21530a54cf196ab00ab7c94ee511164

**Comment by szaimen** [comment]
> > @Zoey2936 would this approach be fine for you?
> 
> what is different to #3096?

Mainly separating the change to caddy from moving to a full-blown apache conf which is better to test

File: Containers/mastercontainer/Caddyfile (hunk 1 above)
Comment (Zoey2936) [reply Containers/mastercontainer/Caddyfile:32]: is this not a global option? https://caddyserver.com/docs/caddyfile/options#trusted-proxies

Comment (szaimen) [reply Containers/mastercontainer/Caddyfile:32]: I think it is both...

Comment (szaimen) [reply Containers/mastercontainer/Caddyfile:32]: https://caddyserver.com/docs/caddyfile/directives/reverse_proxy#the-http-transport

**Review by Zoey2936 — APPROVED** [review]

**Comment by szaimen** [comment]
Thanks :)

**Comment by szaimen** [comment]
I fear this does not work. I get the following issue:

```
automation policy from site block is also default/catch-all policy because of key without hostname, and the two are in conflict: []certmagic.Issuer{(*caddytls.InternalIssuer)(0xc00032c360)} != []certmagic.Issuer{(*caddytls.ACMEIssuer)(0xc0003fce00)}
```

So reverting.

**Comment by szaimen** [comment]
This is now released with v7.4.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Pull request merged**